wordpress-mobile/WordPress-Android
# Integrate ExPlat and add A/A experiment

This PR integrates the ExPlat to the app and adds an A/A experiment implementation so that we can test the integration. 

It builds upon the work previously done on the FluxC library:
- https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1756
- https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1845

For more background context, please check these internal refs:
- pbmo2S-EF-p2
- pbxNRc-Fx-p2

Note: This is ready for review, but not ready for merge just yet in case we need to coordinate with the iOS team.

## To test

This doesn't change anything on the UI, so first make sure all unit tests pass and then, with `logcat` opened, follow the steps below:

1. Open the app.
1. Notice a message like `ExPlat: fetching assignments successful with result`.
1. Put the app on the background and open it again.
1. Make sure you don't see any new message containing `ExPlat`.
1. Close/stop the app and open it again.
1. Notice the message from step 2 appears again.
1. Tap on the Reader button in the bottom navigation bar.
1. Notice a message like `Tracked: reader_accessed`.

**Login/Logout**
1. Log out from the app.
1. Notice a message like `ExPlat: clearing cached assignments`.
1. Login on the app again.
1. Notice a message like `ExPlat: fetching assignments successful with result`. 

---

PR submission checklist:
